prometheus_hardware_exporter/collector.py

```diff
@@ -733,3 +734,113 @@ def fetch(self) -> List[Payload]:
     def process(self, payloads: List[Payload], datastore: Dict[str, Payload]) -> List[Payload]:
         """Process the payload if needed."""
         return payloads
+
+
+class SsaCLICollector(BlockingCollector):
+    """Collector for storage arrays that support ssacli."""
+
+    ssacli = SsaCLI()
+
+    @property
+    def specifications(self) -> List[Specification]:
+        return [
+            Specification(
+                name="ssacli_command_success",
+                documentation="Indicates if the command is successful or not",
+                metric_class=GaugeMetricFamily,
+            ),
+            Specification(
+                name="ssacli_controllers",
+                documentation="Total number of controllers",
+                metric_class=GaugeMetricFamily,
+            ),
+            Specification(
+                name="ssacli_controller_status",
+                documentation="Indicates the status of controller part",
+                labels=["slot", "part"],
+                metric_class=GaugeMetricFamily,
+            ),
+            Specification(
+                name="ssacli_logical_drives",
+                documentation="The number of logical drives in the slot",
+                labels=["slot"],
+                metric_class=GaugeMetricFamily,
+            ),
+            Specification(
+                name="ssacli_physical_drives",
+                documentation="The number of physical drives in the slot",
+                labels=["slot"],
+                metric_class=GaugeMetricFamily,
+            ),
+            Specification(
+                name="ssacli_ld_status",
+                documentation="Indicates the status of logical drive",
+                labels=["slot", "drive_id"],
+                metric_class=GaugeMetricFamily,
+            ),
+            Specification(
+                name="ssacli_pd_status",
+                documentation="Indicates the status of physical drive",
+                labels=["slot", "drive_id"],
+                metric_class=GaugeMetricFamily,
+            ),
+        ]
+
+    def fetch(self) -> List[Payload]:
+        """Load the controller and drive status using ssacli tool."""
+        ssacli_payload = self.ssacli.get_payload()
+
+        if not ssacli_payload:
+            logger.error("Failed to get controllers info using ssacli.")
+            return [Payload(name="ssacli_command_success", value=0.0)]
+        payloads = [
+            Payload(name="ssacli_command_success", value=1.0),
+            Payload(name="ssacli_controllers", value=len(ssacli_payload)),
+        ]
+
+        for slot, payload in ssacli_payload.items():
+            ctrl_status = payload["controller_status"]
+            for part, status in ctrl_status.items():
+                payloads.append(
+                    Payload(
+                        name="ssacli_controller_status",
+                        labels=[slot, part],
+                        value=self._get_status_value(status),
```

Comment:
The design seems different than other collectors. 
https://github.com/canonical/prometheus-hardware-exporter/blob/main/prometheus_hardware_exporter/collector.py#L143

Do we want to align the design ?

Solution 1: state as an info in label
Solution 2: status as a value , 1: healthy, 0: unhealthy